generators/app/templates/.gitignore

```diff
@@ -18,39 +18,32 @@ Thumbs.db
 
 # Compiled Files
 # ---------------------------
-/dist/<% if (platform == 'craft') { %>
+/dist/
 
 # Craft
 # ---------------------------
 /craft/plugins/*
 /craft/rev-manifest.json
-/craft/storage/
+/craft/storage/*
 
 # Craft Public
 # ---------------------------
-/public/.htaccess
-/public/assets/
-/public/assets/**/_*
-/public/images/
-/public/scripts/
-/public/styles/
-/public/uploads/
-/public/web.config<% } %>
+/public/*
+!/public/index.php
+!/public/*.txt
+!/public/*.png
+!/public/*.ico
 
 # Package Managers
 # ---------------------------
-/.bundle
 /vendor/
-bower_components
-node_modules
-npm-debug.log
+/node_modules/
```

Comment:
I feel like we should focus on ignoring files that this generator creates (if we're not ignoring a bunch of common files). Since our projects don't use bower or bundler anymore they seemed unnecessary.